code/modules/xenomorph/xeno_structures.dm

```diff
@@ -452,3 +450,90 @@
 	newshot.permutated += src
 	newshot.def_zone = pick(GLOB.base_miss_chance)
 	newshot.fire_at(hostile, src, null, ammo.max_range, ammo.shell_speed)
+
+/obj/structure/xeno/resin/xeno_turret/proc/turret_message()
+	if(!TIMER_COOLDOWN_CHECK(src, COOLDOWN_XENO_TURRETS_ALERT))
+		associated_hive.xeno_message("Our [name] is attacking a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y]).", "xenoannounce", 5, FALSE, hostile, 'sound/voice/alien_help1.ogg', FALSE, null, /obj/screen/arrow/turret_attacking_arrow)
+		TIMER_COOLDOWN_START(src, COOLDOWN_XENO_TURRETS_ALERT, 50 SECONDS)
+
+//////////////////////
+///// xeno jell turret
+//////////////////////
+
+/obj/structure/xeno/resin/xeno_turret/jelly
+	icon = 'icons/Xeno/jelturret.dmi'
+	icon_state = "jel_turret"
+	name = "Growth jelly turret"
+	desc = "A menacing looking construct of resin, it seems to be alive. It fires larvae growth jelly, useful for breeding dens."
+	///Fire rate of the target in ticks
+	firerate = 20
+
+/obj/structure/xeno/resin/xeno_turret/jelly/Initialize(mapload, hivenumber = XENO_HIVE_NORMAL)
+	.=..()
+	ammo = GLOB.ammo_list[/datum/ammo/xeno/larva_jelly]
+	set_light(2, 2, LIGHT_COLOR_PURPLE)
+
+/obj/structure/xeno/resin/xeno_turret/jelly/process()
+	//Turrets regen some HP, every 2 sec
+	.=..()
+	if(world.time > last_scan_time + TURRET_SCAN_FREQUENCY)
+		scanjel()
+		last_scan_time = world.time
+	if(!hostile || hostile.reagents.has_reagent(/datum/reagent/consumable/larvajelly) || !locate(/obj/item/alien_embryo) in hostile)
+		if(last_hostile)
+			set_last_hostile(null)
+		return
```

Comment:
Can also remove this `return`, since `return` is always implied at the end of a proc. (It'll return no matter what happens.)
```suggestion
		if(last_hostile)
			set_last_hostile(null)
```